src/vm/wren_compiler.c

```diff
@@ -3202,6 +3206,21 @@ static bool method(Compiler* compiler, Variable classVariable)
     createConstructor(compiler, &signature, methodSymbol);
     defineMethod(compiler, classVariable, true, constructorSymbol);
   }
+  
+  // Check for duplicate methods. Doesn't matter that it's already been
+  // defined, error will discard bytecode anyway.
+  IntBuffer* methods = isStatic ? &compiler->enclosingClass->staticMethods
+                                : &compiler->enclosingClass->instanceMethods;
+  
+  int symbol = wrenIntBufferFind(methods, methodSymbol);
+  if (symbol != -1)
+  {
+    const char* methodType = isStatic ? "static method" : "method";
+    error(compiler, "Already defined %s '%s' for class %s in module '%s'.",
+      methodType, fullSignature, &compiler->enclosingClass->name->value,
+      compiler->parser->module->name);
```

Comment:
Ok, I basically copy/pasted most of this out of [an error in vm.c](https://github.com/munificent/wren/blob/master/src/vm/wren_vm.c#L339) which _is_ passing in the module. I hadn't noticed the compiler error function was already doing that.